int32 mode
---
bool ok
